wordpress-mobile/WordPress-Android
# Add domains list card UI

Fixes #19074

### Description

This PR adds composables for the domain card to be displayed in the domain list screen.

Note: In order to utilize the data classes from the API work, this has been pre-merged with https://github.com/wordpress-mobile/WordPress-Android/pull/19360. Some extra commits may appear below, until that PR is merged with a more recent version of `trunk`.

To test:

Check the designs via the Android Studio preview view:

| Preview |  Light Mode | Dark Mode |
|-|-|-|
| Status Rows |  <img width="526" alt="Screenshot 2023-10-13 at 4 59 29 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/1b1793a8-397d-46ff-ba05-3a396f7f003e"> | <img width="531" alt="Screenshot 2023-10-13 at 4 59 47 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/e7890328-5264-4a1b-a7ab-6bbaf7646d6e"> |
| Cards | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/0879d9bc-c8c4-4135-88a4-853b2849ab53) | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/693583ea-b513-4e78-b068-048146eb5bd5) |






## Regression Notes
1. Potential unintended areas of impact
N/A (The composables are not yet in use, except for in previews, i.e. no app usage yet).

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
